test/test-trace-writer.js

```diff
@@ -18,94 +18,133 @@
 
 var assert = require('assert');
 var nock = require('nock');
-var cls = require('../src/cls.js');
-var common = require('./plugins/common.js');
-var trace = require('..');
+var nocks = require('./nocks.js');
+var Service = require('@google-cloud/common').Service;
+var traceLabels = require('../src/trace-labels.js');
 
 nock.disableNetConnect();
 
-var uri = 'https://cloudtrace.googleapis.com';
-var path = '/v1/projects/0/traces';
+var PROJECT = 'fake-project';
+var DEFAULT_DELAY = 200;
 
-process.env.GCLOUD_PROJECT = 0;
 
-var queueSpans = function(n, agent) {
-  for (var i = 0; i < n; i++) {
-    common.runInTransaction(agent, function(end) {
-      end();
-    });
-  }
+var fakeLogger = {
+  warn: function() {},
+  info: function() {},
+  error: function() {},
+  debug: function() {}
 };
 
-describe('tracewriter publishing', function() {
-
-  it('should publish when queue fills', function(done) {
-    var buf;
-    var scope = nock(uri)
-        .intercept(path, 'PATCH', function(body) {
-          assert.equal(JSON.stringify(body.traces), JSON.stringify(buf));
-          return true;
-        }).reply(200);
-    var agent = trace.start({
-      bufferSize: 2,
-      samplingRate: 0,
-      forceNewAgent_: true
-    });
-    common.avoidTraceWriterAuth(agent);
-    cls.getNamespace().run(function() {
-      queueSpans(2, agent);
-      buf = common.getTraces(agent);
-      setTimeout(function() {
-        scope.done();
+function createFakeSpan(name) {
```

Comment:
Maybe. Depends on whether the other tests want the mock to be more functional. It is a tradeoff between what this set of unit tests needs vs reimplementing all the functionality of `SpanData`. `SpanData` has dependencies on `Agent` and `Logger` which need to be injected, perhaps it is better to inject these dependencies and use the real `SpanData` class?